app_dart/lib/src/service/scheduler.dart

```diff
@@ -183,9 +191,17 @@ class Scheduler {
   /// Load in memory the `.ci.yaml`.
   Future<CiYaml> getCiYaml(
     Commit commit, {
+    CiYaml? totCiYaml,
     RetryOptions retryOptions = const RetryOptions(maxAttempts: 3),
   }) async {
-    final String ciPath = '${commit.repository}/${commit.sha!}/$kCiYamlPath';
+    String ciPath;
+
+    /// Use default branch instead of sha when pulling against tip of tree.
+    if (commit.sha == null) {
```

Comment:
`commit.sha` should in practice never be null. Can we throw an exception instead?

The gcloud datastore entities in dart require all fields to be nullable so they can be lazily set, but by the time they're used in cocoon it's non-nullable